Content.Client/NukeOps/WarDeclaratorWindow.xaml.cs

```diff
@@ -128,4 +132,18 @@ public void UpdateTimer()
                 return;
         }
     }
+
+    private void HandleTextChanged(TextEdit.TextEditEventArgs args)
+    {
+        var length = MessageEdit.TextLength;
+        var maxLength = _cfg.GetCVar(CCVars.ChatMaxAnnouncementLength);
+        var remaining = maxLength - length;
+
+        MessageLengthLabel.Visible = remaining <= 20;
+        MessageLengthLabel.Text = $"{remaining}";
+        MessageLengthLabel.FontColorOverride = remaining < 0 ? Color.Red : null;
+        MessageLengthLabel.ToolTip = remaining < 0
+            ? _loc.GetString("war-declarator-message-length-too-long-tooltip")
+            : _loc.GetString("war-declarator-message-length-remaining-tooltip", ("remaining", remaining));
+    }
 }
```

Comment:
Code should be shared with comms console as right now they just copy-paste each other.

Reply:
@metalgearsloth, can you take another look, please?

---

I was aware there is a code duplication. I stated it right in the PR's description. I just wanted an opinion on whenever introducing another UI control is a good idea or not. Anyway, I committed a reusable control that should address code duplication.